### PR TITLE
*.yml: move dhcpcd from onboot to services

### PR DESCRIPTION
**Pull request by AkihiroSuda**
**- What I did**
move dhcpcd from onboot to services, so that the instance can invoke DHCP request on network reconnection and on lease timeout

**- How I did it**
Edit *.yml

**- How to verify it**
```
host$ moby build minimal.yml && linuxkit run minimal
guest$ runc list
ID          PID         STATUS      BUNDLE                        CREATED                          OWNER           
dhcpcd      301         running     /containers/services/dhcpcd   2017-04-25T05:13:03.762592514Z   root
```

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

penguin

Signed-off-by: Akihiro Suda <suda.akihiro@lab.ntt.co.jp>

### CODE REVIEW

**Comment by rneugeba** [comment]
The reason for invoking `dhcp` in onboot is so that we can guarantee that we have an IP address for everything starting afterwards. If we just start dhcp in the service section other services, like for example an nginx contianer, need to check/wait till the network is up. We had discussed to use dhcpd in oneshot mode in `onboot` *and* as a service

**Comment by rneugeba** [comment]
This commit also seems to have other unrelated changes, like changing the SHA for init and containerd. Please don't mix them in one commit and provide them in a separate commit with an appropriate commit message

File: examples/minimal.yml [hunk 1]
```diff
@@ -2,12 +2,12 @@ kernel:
   image: "linuxkit/kernel:4.9.x"
   cmdline: "console=ttyS0 console=tty0 page_poison=1"
 init:
-  - linuxkit/init:42fe8cb1508b3afed39eb89821906e3cc7a70551
+  - linuxkit/init:63eed9ca7a09d2ce4c0c5e7238ac005fa44f564b
   - linuxkit/runc:b0fb122e10dbb7e4e45115177a61a3f8d68c19a9
-  - linuxkit/containerd:60e2486a74c665ba4df57e561729aec20758daed
-onboot:
+  - linuxkit/containerd:18eaf72f3f4f9a9f29ca1951f66df701f873060b
```
Comment (AkihiroSuda) [inline examples/minimal.yml]: Updating containerd (and init to the corresponding version) here is intentional. The old version does not start up for some reason:
```
/ # cat /var/log/dhcpcd.log
ctr: rpc error: code = 14 desc = grpc: the connection is unavailable
```

**Comment by AkihiroSuda** [comment]
@rneugeba 

Thank you for the information.
Before duplicating dhcpd on-boots in all the yaml files and keeping them up-to-date, I think we should better implement https://github.com/moby/tool/issues/5 (inheritance) or https://github.com/moby/tool/issues/4 (include) in the builder.

So I'm temporarily closing this PR.

As mentioned above, other modifications are intentional, but I'll investigate and fix up in another issue/PR.

**Comment by rneugeba** [comment]
Thanks. the `init` version needs to be matched with the `runc` and `containerd` version. There is at least one version of `init` which starts a hacked version of `containerd` instead of using `runc`. The hacked version of `containerd` has a modification to `ctr` to use a `runc` style rootfs but has some issues with logging and seems to be racey during start (#1660, #1661). We added it before DockerCon, so we could do `ctr list` for the demo. The plan is to create a `containerd` style imagestore with `moby` to properly (and fix the `containerd` issues.

**Pull request closed without merging**